Collatz.java

```diff
@@ -1,6 +1,53 @@
-// Demonstrates the Collatz conjecture.
+
 public class Collatz {
 	public static void main(String args[]) {
-	    // Replace this comment with your code
+		int numOfTimes = Integer.parseInt(args[0]);
+		String type = args[1];
+		boolean reach = false;
+		if (type.equals("v")) {
+			for (int j = 1; j <= numOfTimes; j++) {
+				int counter = 1;
+				int current = j;
+				reach = false;
+				String halistone = "";
+				while (reach == false) {
+					System.out.print(current + " ");
+					if (current % 2 == 0){
+						current = current / 2;			
+					}
+					else {
+						current = (current * 3) + 1;		
+					}
+					counter++;
+					if (current == 1){
+						System.out.print(current + " (" + counter + ")");
+						reach = true;
+						System.out.println();
+					}
+				}
+				
+			}
+		}
+		else {
```

Comment:
Since the _concise_ mode is almost the same as the _verbose_ mode (they just differ in a couple of lines of code) maybe you can combine them in a way that you won't need to write the same function twice :)